### ISSUE TITLE
SchedulerPredicates validates that NodeSelector is respected if matching flaky

### ISSUE THREAD

**Issue by dchen1107** [kubernetes/kubernetes#15575]
Jenkins project kubernetes-e2e-gke-ci build: http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gke-ci/8884/

```
chedulerPredicates validates that NodeSelector is respected if matching [Conformance]

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:449
Expected error:
    <*errors.errorString | 0xc208962760>: {
        s: "Giving up; pod went into failed status: \n(*api.Pod){TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)with-labels GenerateName:(string) Namespace:(string)e2e-tests-sched-pred-3pr59 SelfLink:(string)/api/v1/namespaces/e2e-tests-sched-pred-3pr59/pods/with-labels UID:(types.UID)2c7f0fae-71f0-11e5-9b0a-42010af00002 ResourceVersion:(string)4106 Generation:(int64)0 CreationTimestamp:(unversioned.Time)2015-10-13 14:20:05 -0700 PDT DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PodSpec){Volumes:([]api.Volume)[{Name:(string)default-token-23l0z VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)<nil> GitRepo:(*api.GitRepoVolumeSource)<nil> Secret:(*api.SecretVolumeSource){SecretName:(string)default-token-23l0z} NFS:(*api.NFSVolumeSource)<nil> ISCSI:(*api.ISCSIVolumeSource)<nil> Glusterfs:(*api.GlusterfsVolumeSource)<nil> PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)<nil> RBD:(*api.RBDVolumeSource)<nil> Cinder:(*api.CinderVolumeSource)<nil> CephFS:(*api.CephFSVolumeSource)<nil> Flocker:(*api.FlockerVolumeSource)<nil> DownwardAPI:(*api.DownwardAPIVolumeSource)<nil> FC:(*api.FCVolumeSource)<nil>}}] Containers:([]api.Container)[{Name:(string)with-labels Image:(string)beta.gcr.io/google_containers/pause:2.0 Command:([]string)<nil> Args:([]string)<nil> WorkingDir:(string) Ports:([]api.ContainerPort)<nil> Env:([]api.EnvVar)<nil> Resources:(api.ResourceRequirements){Limits:(api.ResourceList)<nil> Requests:(api.ResourceList)<nil>} VolumeMounts:([]api.VolumeMount)[{Name:(string)default-token-23l0z ReadOnly:(bool)true MountPath:(string)/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:(*api.Probe)<nil> ReadinessProbe:(*api.Probe)<nil> Lifecycle:(*api.Lifecycle)<nil> TerminationMessagePath:(string)/dev/termination-log ImagePullPolicy:(api.PullPolicy)IfNotPresent SecurityContext:(*api.SecurityContext)<nil> Stdin:(bool)false TTY:(bool)false}] RestartPolicy:(api.RestartPolicy)Always TerminationGracePeriodSeconds:(*int64)30 ActiveDeadlineSeconds:(*int64)<nil> DNSPolicy:(api.DNSPolicy)ClusterFirst NodeSelector:(map[string]string)map[kubernetes.io/hostname:gke-jkns-gke-e2e-ci-c036f29e-node-i20n kubernetes.io/e2e-2c7d9913-71f0-11e5-ab00-42010af01555:42] ServiceAccountName:(string)default NodeName:(string)gke-jkns-gke-e2e-ci-c036f29e-node-i20n SecurityContext:(*api.PodSecurityContext){HostNetwork:(bool)false HostPID:(bool)false HostIPC:(bool)false} ImagePullSecrets:([]api.LocalObjectReference)<nil>} Status:(api.PodStatus){Phase:(api.PodPhase)Failed Conditions:([]api.PodCondition)<nil> Message:(string)Pod cannot be started due to node selector mismatch Reason:(string)NodeSelectorMismatching HostIP:(string) PodIP:(string) StartTime:(*unversioned.Time)2015-10-13 14:20:05 -0700 PDT ContainerStatuses:([]api.ContainerStatus)<nil>}}",
    }
    Giving up; pod went into failed status: 
    (*api.Pod){TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)with-labels GenerateName:(string) Namespace:(string)e2e-tests-sched-pred-3pr59 SelfLink:(string)/api/v1/namespaces/e2e-tests-sched-pred-3pr59/pods/with-labels UID:(types.UID)2c7f0fae-71f0-11e5-9b0a-42010af00002 ResourceVersion:(string)4106 Generation:(int64)0 CreationTimestamp:(unversioned.Time)2015-10-13 14:20:05 -0700 PDT DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PodSpec){Volumes:([]api.Volume)[{Name:(string)default-token-23l0z VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)<nil> GitRepo:(*api.GitRepoVolumeSource)<nil> Secret:(*api.SecretVolumeSource){SecretName:(string)default-token-23l0z} NFS:(*api.NFSVolumeSource)<nil> ISCSI:(*api.ISCSIVolumeSource)<nil> Glusterfs:(*api.GlusterfsVolumeSource)<nil> PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)<nil> RBD:(*api.RBDVolumeSource)<nil> Cinder:(*api.CinderVolumeSource)<nil> CephFS:(*api.CephFSVolumeSource)<nil> Flocker:(*api.FlockerVolumeSource)<nil> DownwardAPI:(*api.DownwardAPIVolumeSource)<nil> FC:(*api.FCVolumeSource)<nil>}}] Containers:([]api.Container)[{Name:(string)with-labels Image:(string)beta.gcr.io/google_containers/pause:2.0 Command:([]string)<nil> Args:([]string)<nil> WorkingDir:(string) Ports:([]api.ContainerPort)<nil> Env:([]api.EnvVar)<nil> Resources:(api.ResourceRequirements){Limits:(api.ResourceList)<nil> Requests:(api.ResourceList)<nil>} VolumeMounts:([]api.VolumeMount)[{Name:(string)default-token-23l0z ReadOnly:(bool)true MountPath:(string)/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:(*api.Probe)<nil> ReadinessProbe:(*api.Probe)<nil> Lifecycle:(*api.Lifecycle)<nil> TerminationMessagePath:(string)/dev/termination-log ImagePullPolicy:(api.PullPolicy)IfNotPresent SecurityContext:(*api.SecurityContext)<nil> Stdin:(bool)false TTY:(bool)false}] RestartPolicy:(api.RestartPolicy)Always TerminationGracePeriodSeconds:(*int64)30 ActiveDeadlineSeconds:(*int64)<nil> DNSPolicy:(api.DNSPolicy)ClusterFirst NodeSelector:(map[string]string)map[kubernetes.io/hostname:gke-jkns-gke-e2e-ci-c036f29e-node-i20n kubernetes.io/e2e-2c7d9913-71f0-11e5-ab00-42010af01555:42] ServiceAccountName:(string)default NodeName:(string)gke-jkns-gke-e2e-ci-c036f29e-node-i20n SecurityContext:(*api.PodSecurityContext){HostNetwork:(bool)false HostPID:(bool)false HostIPC:(bool)false} ImagePullSecrets:([]api.LocalObjectReference)<nil>} Status:(api.PodStatus){Phase:(api.PodPhase)Failed Conditions:([]api.PodCondition)<nil> Message:(string)Pod cannot be started due to node selector mismatch Reason:(string)NodeSelectorMismatching HostIP:(string) PodIP:(string) StartTime:(*unversioned.Time)2015-10-13 14:20:05 -0700 PDT ContainerStatuses:([]api.ContainerStatus)<nil>}}
not to have occurred
```

One more flakiness blocks entire submit queue.

**Comment by dchen1107** [comment]
cc/ @sttts

**Comment by sttts** [comment]
Looking into this.

**Comment by sttts** [comment]
@dchen1107 do you have a longer test trace for me? 

I cannot see much that could be flakey due to the tests. It just does this:

- start pod without-label
- remember NodeName
- delete without-label
- set label kubernetes.io/e2e-<some-uuid>=42 on NodeName
- launch pod with-label and NodeSelector set to {"kubernetes.io/hostname": nodeName, kubernetes.io/e2e-<some-uuid>":42}

The only idea I have is that the patch command does not work reliably:
```
patch := fmt.Sprintf(`{"metadata":{"labels":{"%s":"%s"}}}`, k, v)
err = c.Patch(api.MergePatchType).Resource("nodes").Name(nodeName).Body([]byte(patch)).Do().Error()
expectNoError(err)
```

**Comment by dchen1107** [comment]
@sttts here is the most recent failure from our Jenkins build:

```
15:00:57 SchedulerPredicates 
15:00:57   validates that NodeSelector is respected if matching [Conformance]
15:00:57   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:449
15:00:57 [BeforeEach] SchedulerPredicates
15:00:57   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:176
15:00:57 STEP: Creating a kubernetes client
15:00:57 >>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gke-ci/workspace/.kube/config
15:00:57 STEP: Building a namespace api object
15:00:57 Oct 15 15:00:57.772: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-sched-pred-eupkg
15:00:57 Oct 15 15:00:57.774: INFO: Get service account default in ns e2e-tests-sched-pred-eupkg failed, ignoring for 2s: serviceaccounts "default" not found
15:00:59 Oct 15 15:00:59.777: INFO: Service account default in ns e2e-tests-sched-pred-eupkg with secrets found. (2.00436108s)
15:00:59 STEP: Waiting for a default service account to be provisioned in namespace
15:00:59 Oct 15 15:00:59.777: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-sched-pred-eupkg
15:00:59 Oct 15 15:00:59.779: INFO: Service account default in ns e2e-tests-sched-pred-eupkg with secrets found. (1.689522ms)
15:00:59 [It] validates that NodeSelector is respected if matching [Conformance]
15:00:59   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:449
15:00:59 STEP: Trying to launch a pod without a label to get a node which can launch it.
15:00:59 Oct 15 15:00:59.786: INFO: Waiting up to 5m0s for pod without-label status to be running
15:00:59 Oct 15 15:00:59.788: INFO: Waiting for pod without-label in namespace 'e2e-tests-sched-pred-eupkg' status to be 'running'(found phase: "Pending", readiness: false) (1.901506ms elapsed)
15:01:01 Oct 15 15:01:01.792: INFO: Found pod 'without-label' on node 'gke-jkns-gke-e2e-ci-281f8295-node-ga3m'
15:01:01 STEP: Trying to apply a random label on the found node.
15:01:01 STEP: Trying to relaunch the pod, now with labels.
15:01:01 Oct 15 15:01:01.815: INFO: Waiting up to 5m0s for pod with-labels status to be running
15:01:01 Oct 15 15:01:01.817: INFO: Waiting for pod with-labels in namespace 'e2e-tests-sched-pred-eupkg' status to be 'running'(found phase: "Pending", readiness: false) (1.925199ms elapsed)
15:01:03 [AfterEach] SchedulerPredicates
15:01:03   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:186
15:01:03 STEP: Collecting events from namespace "e2e-tests-sched-pred-eupkg".
15:01:03 Oct 15 15:01:03.838: INFO: event for with-labels: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} NodeSelectorMismatching: cannot be started due to node selector mismatch
15:01:03 Oct 15 15:01:03.838: INFO: event for with-labels: {scheduler } Scheduled: Successfully assigned with-labels to gke-jkns-gke-e2e-ci-281f8295-node-ga3m
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Pulled: Container image "beta.gcr.io/google_containers/pause:2.0" already present on machine
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {scheduler } Scheduled: Successfully assigned without-label to gke-jkns-gke-e2e-ci-281f8295-node-ga3m
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Created: Created with docker id b03c0b825d89
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Started: Started with docker id b03c0b825d89
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Created: Created with docker id 0c70c2f8abba
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Started: Started with docker id 0c70c2f8abba
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Killing: Killing with docker id 0c70c2f8abba
15:01:03 Oct 15 15:01:03.838: INFO: event for without-label: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} Killing: Killing with docker id b03c0b825d89
15:01:03 Oct 15 15:01:03.845: INFO: POD                                                           NODE                                    PHASE    GRACE  CONDITIONS
15:01:03 Oct 15 15:01:03.845: INFO: fluentd-cloud-logging-gke-jkns-gke-e2e-ci-281f8295-node-ga3m  gke-jkns-gke-e2e-ci-281f8295-node-ga3m  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-15 14:57:15 -0700 PDT  }]
15:01:03 Oct 15 15:01:03.845: INFO: fluentd-cloud-logging-gke-jkns-gke-e2e-ci-281f8295-node-pphb  gke-jkns-gke-e2e-ci-281f8295-node-pphb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-15 14:57:20 -0700 PDT  }]
15:01:03 Oct 15 15:01:03.845: INFO: heapster-v10-6kujc                                            gke-jkns-gke-e2e-ci-281f8295-node-pphb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-15 14:57:02 -0700 PDT  }]
15:01:03 Oct 15 15:01:03.845: INFO: kube-dns-v9-yqvfq                                             gke-jkns-gke-e2e-ci-281f8295-node-pphb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-15 14:57:13 -0700 PDT  }]
15:01:03 Oct 15 15:01:03.845: INFO: kube-ui-v3-fxn44                                              gke-jkns-gke-e2e-ci-281f8295-node-pphb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-15 14:56:56 -0700 PDT  }]
15:01:03 Oct 15 15:01:03.845: INFO: 
15:01:03 Oct 15 15:01:03.845: INFO: Waiting up to 1m0s for all nodes to be ready
15:01:03 Oct 15 15:01:03.848: INFO: Node gke-jkns-gke-e2e-ci-281f8295-node-ga3m condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-15 14:56:53 -0700 PDT
15:01:03 Oct 15 15:01:03.848: INFO: Successfully found node gke-jkns-gke-e2e-ci-281f8295-node-ga3m readiness to be true
15:01:03 Oct 15 15:01:03.848: INFO: Node gke-jkns-gke-e2e-ci-281f8295-node-pphb condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-15 14:56:51 -0700 PDT
15:01:03 Oct 15 15:01:03.848: INFO: Successfully found node gke-jkns-gke-e2e-ci-281f8295-node-pphb readiness to be true
15:01:03 STEP: Destroying namespace "e2e-tests-sched-pred-eupkg" for this suite.
15:01:08 
15:01:08 • Failure [11.093 seconds]
15:01:08 SchedulerPredicates
15:01:08 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:450
15:01:08   validates that NodeSelector is respected if matching [Conformance] [It]
15:01:08   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:449
15:01:08 
15:01:08   Expected error:
15:01:08       <*errors.errorString | 0xc2082d6ef0>: {
15:01:08           s: "Giving up; pod went into failed status: \n(*api.Pod){TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)with-labels GenerateName:(string) Namespace:(string)e2e-tests-sched-pred-eupkg SelfLink:(string)/api/v1/namespaces/e2e-tests-sched-pred-eupkg/pods/with-labels UID:(types.UID)39474cf2-7388-11e5-bf19-42010af00002 ResourceVersion:(string)294 Generation:(int64)0 CreationTimestamp:(unversioned.Time)2015-10-15 15:01:01 -0700 PDT DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PodSpec){Volumes:([]api.Volume)[{Name:(string)default-token-qs050 VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)<nil> GitRepo:(*api.GitRepoVolumeSource)<nil> Secret:(*api.SecretVolumeSource){SecretName:(string)default-token-qs050} NFS:(*api.NFSVolumeSource)<nil> ISCSI:(*api.ISCSIVolumeSource)<nil> Glusterfs:(*api.GlusterfsVolumeSource)<nil> PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)<nil> RBD:(*api.RBDVolumeSource)<nil> Cinder:(*api.CinderVolumeSource)<nil> CephFS:(*api.CephFSVolumeSource)<nil> Flocker:(*api.FlockerVolumeSource)<nil> DownwardAPI:(*api.DownwardAPIVolumeSource)<nil> FC:(*api.FCVolumeSource)<nil>}}] Containers:([]api.Container)[{Name:(string)with-labels Image:(string)beta.gcr.io/google_containers/pause:2.0 Command:([]string)<nil> Args:([]string)<nil> WorkingDir:(string) Ports:([]api.ContainerPort)<nil> Env:([]api.EnvVar)<nil> Resources:(api.ResourceRequirements){Limits:(api.ResourceList)<nil> Requests:(api.ResourceList)<nil>} VolumeMounts:([]api.VolumeMount)[{Name:(string)default-token-qs050 ReadOnly:(bool)true MountPath:(string)/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:(*api.Probe)<nil> ReadinessProbe:(*api.Probe)<nil> Lifecycle:(*api.Lifecycle)<nil> TerminationMessagePath:(string)/dev/termination-log ImagePullPolicy:(api.PullPolicy)IfNotPresent SecurityContext:(*api.SecurityContext)<nil> Stdin:(bool)false TTY:(bool)false}] RestartPolicy:(api.RestartPolicy)Always TerminationGracePeriodSeconds:(*int64)30 ActiveDeadlineSeconds:(*int64)<nil> DNSPolicy:(api.DNSPolicy)ClusterFirst NodeSelector:(map[string]string)map[kubernetes.io/hostname:gke-jkns-gke-e2e-ci-281f8295-node-ga3m kubernetes.io/e2e-3941e1a5-7388-11e5-8fbe-42010af01555:42] ServiceAccountName:(string)default NodeName:(string)gke-jkns-gke-e2e-ci-281f8295-node-ga3m SecurityContext:(*api.PodSecurityContext){HostNetwork:(bool)false HostPID:(bool)false HostIPC:(bool)false} ImagePullSecrets:([]api.LocalObjectReference)<nil>} Status:(api.PodStatus){Phase:(api.PodPhase)Failed Conditions:([]api.PodCondition)<nil> Message:(string)Pod cannot be started due to node selector mismatch Reason:(string)NodeSelectorMismatching HostIP:(string) PodIP:(string) StartTime:(*unversioned.Time)2015-10-15 15:01:01 -0700 PDT ContainerStatuses:([]api.ContainerStatus)<nil>}}",
15:01:08       }
15:01:08       Giving up; pod went into failed status: 
15:01:08       (*api.Pod){TypeMeta:(unversioned.TypeMeta){Kind:(string) APIVersion:(string)} ObjectMeta:(api.ObjectMeta){Name:(string)with-labels GenerateName:(string) Namespace:(string)e2e-tests-sched-pred-eupkg SelfLink:(string)/api/v1/namespaces/e2e-tests-sched-pred-eupkg/pods/with-labels UID:(types.UID)39474cf2-7388-11e5-bf19-42010af00002 ResourceVersion:(string)294 Generation:(int64)0 CreationTimestamp:(unversioned.Time)2015-10-15 15:01:01 -0700 PDT DeletionTimestamp:(*unversioned.Time)<nil> DeletionGracePeriodSeconds:(*int64)<nil> Labels:(map[string]string)<nil> Annotations:(map[string]string)<nil>} Spec:(api.PodSpec){Volumes:([]api.Volume)[{Name:(string)default-token-qs050 VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSource)<nil> GitRepo:(*api.GitRepoVolumeSource)<nil> Secret:(*api.SecretVolumeSource){SecretName:(string)default-token-qs050} NFS:(*api.NFSVolumeSource)<nil> ISCSI:(*api.ISCSIVolumeSource)<nil> Glusterfs:(*api.GlusterfsVolumeSource)<nil> PersistentVolumeClaim:(*api.PersistentVolumeClaimVolumeSource)<nil> RBD:(*api.RBDVolumeSource)<nil> Cinder:(*api.CinderVolumeSource)<nil> CephFS:(*api.CephFSVolumeSource)<nil> Flocker:(*api.FlockerVolumeSource)<nil> DownwardAPI:(*api.DownwardAPIVolumeSource)<nil> FC:(*api.FCVolumeSource)<nil>}}] Containers:([]api.Container)[{Name:(string)with-labels Image:(string)beta.gcr.io/google_containers/pause:2.0 Command:([]string)<nil> Args:([]string)<nil> WorkingDir:(string) Ports:([]api.ContainerPort)<nil> Env:([]api.EnvVar)<nil> Resources:(api.ResourceRequirements){Limits:(api.ResourceList)<nil> Requests:(api.ResourceList)<nil>} VolumeMounts:([]api.VolumeMount)[{Name:(string)default-token-qs050 ReadOnly:(bool)true MountPath:(string)/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:(*api.Probe)<nil> ReadinessProbe:(*api.Probe)<nil> Lifecycle:(*api.Lifecycle)<nil> TerminationMessagePath:(string)/dev/termination-log ImagePullPolicy:(api.PullPolicy)IfNotPresent SecurityContext:(*api.SecurityContext)<nil> Stdin:(bool)false TTY:(bool)false}] RestartPolicy:(api.RestartPolicy)Always TerminationGracePeriodSeconds:(*int64)30 ActiveDeadlineSeconds:(*int64)<nil> DNSPolicy:(api.DNSPolicy)ClusterFirst NodeSelector:(map[string]string)map[kubernetes.io/hostname:gke-jkns-gke-e2e-ci-281f8295-node-ga3m kubernetes.io/e2e-3941e1a5-7388-11e5-8fbe-42010af01555:42] ServiceAccountName:(string)default NodeName:(string)gke-jkns-gke-e2e-ci-281f8295-node-ga3m SecurityContext:(*api.PodSecurityContext){HostNetwork:(bool)false HostPID:(bool)false HostIPC:(bool)false} ImagePullSecrets:([]api.LocalObjectReference)<nil>} Status:(api.PodStatus){Phase:(api.PodPhase)Failed Conditions:([]api.PodCondition)<nil> Message:(string)Pod cannot be started due to node selector mismatch Reason:(string)NodeSelectorMismatching HostIP:(string) PodIP:(string) StartTime:(*unversioned.Time)2015-10-15 15:01:01 -0700 PDT ContainerStatuses:([]api.ContainerStatus)<nil>}}
15:01:08   not to have occurred
15:01:08 
15:01:08   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:445
```

**Comment by sttts** [comment]
@dchen1107 thanks for the logs.

This clarifies what happens: the kubelet does not know its new labels. Maybe, its node storage didn't update fast enough and there is a race with the scheduler:
```
15:01:03 Oct 15 15:01:03.838: INFO: event for with-labels: {kubelet gke-jkns-gke-e2e-ci-281f8295-node-ga3m} NodeSelectorMismatching: cannot be started due to node selector mismatch
```

**Comment by sttts** [comment]
This might be fixed with #15772.
